### PR TITLE
#2479: setup and migrate fact data to date2serial lookup table

### PR DESCRIPTION
**Pull request by uweschaefer**
that'll be a tough review :rofl:

### CODE REVIEW

**Comment by uweschaefer** [comment]
closes #2479

File: .github/workflows/maven.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
     steps:
     - id: 'step-0'
       name: 'Checkout'
-      uses: 'actions/checkout@v4'
```
Comment (bresciamattia) [inline .github/workflows/maven.yaml:17]: is this intended?

File: .github/workflows/maven.yaml [hunk 2]
```diff
@@ -59,15 +59,14 @@ jobs:
...
-        - '11'
         - '12'
         - '13'
         - '14'
         - '15'
     steps:
     - id: 'step-0'
       name: 'Checkout'
-      uses: 'actions/checkout@v4'
+      uses: 'actions/checkout@v3'
```
Comment (bresciamattia) [inline .github/workflows/maven.yaml:69]: same as above

**Review by bresciamattia — APPROVED** [review]
thanks @uweschaefer, lgtm :+1:

**Pull request merged**